app/src/test/java/com/mstudio/superheroarch/presentation/MainViewModelTest.kt

```diff
@@ -76,4 +77,22 @@ class MainViewModelTest {
 
         verify(view).goToDetailScreen(characterSelected)
     }
+
+    @Test
+    fun `given main screen, when user clicks on favorites button and has favorite characters, then show all the favorite characters`() = runTest {
+        val expectedCharacters = listOf(RickAndMortyRepositoryInstruments.givenACharacterEntity(isFavorite = true).toCharacterData())
+        `when`(useCase.getAllCharacters()).thenReturn(expectedCharacters)
+        viewModel.onCreate()
+        viewModel.onFavoriteButtonClicked()
+
+        verify(view, times(2)).showCharacters(expectedCharacters.filter { it.isFavorite })
```

Comment:
this times 2 seems a bit weird, like if you are putting it to make the test work. Maybe there is another way to do this test or the asserttion

Reply:
actually in the next one you are not calling the onCreate, the scenario should be similar but returning empty list

Reply:
I discover the mistery, the other tests wasn't tested properly, now the other tests are correctly implemented and the other about filters is asking for times. because verify has times(1) as default and if it's called more than one fails if you don't put the times, I've changed for atLeastOnce()